CIP-1694/README.md

```diff
@@ -0,0 +1,624 @@
+
+// TODO: Provide a CBOR specification in the annexe for this new on-chain entity
+
+Additionally, the action will include the following:
+
+1. For protocol parameter changes - the changed parameters;
+2. For hard fork initiation - the new major protocol version, which must be one greater than the current version;
+3. For treasury withdrawals - a map from stake credentials to a positive number of Lovelace;
+4. For updates to the Constitution - a 32-byte blake2b-256 hash digest of the Constitution document;
+5. For new Constitutional Committee and changes to the quorum size - a set of key hashes and a positive number that is no greater than the size of the committee;
+6. For a motion of no confidence - nothing else.
+
+The deposit amount will be added to the _deposit pot_, similar to stake key deposits.
+
+#### Governance Action IDs
+
+Each accepted governance action will be assigned a unique identifier (a.k.a. the **governance action ID**), consisting of the transaction ID that created it and the index within the transaction body that points to it.
+
+### Votes
+
+Each vote transaction consists of the following:
+
+* a governance action ID;
+* a role - Constitutional Committee, DRep, or SPO;
+* a key-hash (which will be verified to have the role above);
+* a URL for any metadata that is relevant to the vote;
+* a hash of the contents of this URL;
+* a yes/no/abstain vote.
+
+Note that "abstained" votes are not included in the "active voting stake".
+Note also that a vote to abstain is different than abstaining from voting
+(the former is visible on chain, the latter is not).
+To avoid confusion, we will only use the word "abstain" from this point onward to mean a vote
+to abstain.
+
+The key hash will trigger the appropriate signature check on the transaction body according to the existing `UTxOW` ledger rule.
+
+Votes can be cast multiple times for each governance action by a single key hash. Correctly submitted votes override any older votes for the same key hash and role.
+As soon as a governance action is ratified, voting ends. Any future votes are not considered or recorded (whether they are yes, no or abstain).
+
+#### Governance State
+
+When a governance action is successfully submitted to the chain, its progress will be tracked by the ledger state.
+In particular, the following will be tracked:
+
+* the governance action ID;
+* the epoch that the action expires;
+* the deposit amount;
+* the rewards address that will receive the deposit when it is returned;
+* the total yes/no/abstain votes of the Constitutional Committee for this action;
+* the total yes/no/abstain votes of the DReps for this action;
+* the total yes/no/abstain votes of the SPOs  for this action.
+
+#### Stale Votes
+
+The votes from the DReps and the SPOs may become meaningless after crossing an epoch boundary since they may become unregistered. Therefore, all unregistered votes are cancelled before any new votes are considered.
```

Comment:
> Is it because the stake would change between epochs

yes, exactly.

> What does it mean that all "unregistered" votes are cancelled?

just that they now have a zero weight.

> What is the difference between registered and unregistered votes in this context?

nothing really, 0 is really just as special case of the stake distribution changing. (though there will be a mechanism for un-registering).

Reply:
ok, makes sense, thanks